### PR TITLE
feat(trial balance) display account label on the errors grid

### LINKED ISSUE #3534: Trial Balance should tell which account/debtor group/line of the transation has an error
The Trial Balance does not tell me what line in the Posting Journal has an error.  See below:

incoherenterrormessages
_Fig 1: Incomplete error messages on the Trial Balance_

This makes it really hard to figure out what is going wrong.  We should at least tell the user the name of the debtor group, the name of the account, etc, so they can verify it.

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #3534

### CODE REVIEW

**Comment by jniles** [comment]
Can you post a screenshot?

**Comment by jeremielodi** [comment]
errors

**Review by jniles — COMMENTED** [review]
@jeremielodi, this is close.

For each error, we want to say what the error is and how to fix it.  So, basically:
 1. For "Error: locked fiscal year", we want to say "Fiscal year {{fiscalYearLabel}} is locked and must be reopened before data is posted to it."
 2. For "Error: Users is Locked", we want to say "The user {{username}} has a locked account.  Please reopen {{username}} to allow them to post records."
 3. For Unbalanced transactions, we want to say "This transactions has debits equal to {{sumdebits}} and credits equal to {{sumcredits}}.  All transactions must balance before being posted."
... etc.

That is what the goal of https://github.com/IMA-WorldHealth/bhima/issues/3534 was.

**Comment by jniles** [comment]
@jeremielodi, what is the status of this PR?

**Comment by jeremielodi** [comment]
@jniles, This PR is ready for a review.

File: server/models/procedures/trial_balance.sql [hunk 1]
```diff
@@ -96,91 +96,107 @@ CREATE PROCEDURE TrialBalanceErrors()
...
-    SELECT pj.record_uuid, MAX(pj.trans_id), 'POSTING_JOURNAL.ERRORS.SINGLE_LINE_TRANSACTION' AS code
+    SELECT pj.record_uuid, MAX(pj.trans_id), ac.id, pj.trans_id  as error_description, 'POSTING_JOURNAL.ERRORS.SINGLE_LINE_TRANSACTION' AS code
     FROM posting_journal AS pj JOIN stage_trial_balance_transaction AS temp
       ON pj.record_uuid = temp.record_uuid
+      JOIN account AS ac ON ac.id = pj.account_id
     GROUP BY pj.record_uuid
     HAVING COUNT(pj.record_uuid) < 2;
 
-  SELECT DISTINCT BUID(record_uuid) AS record_uuid, trans_id, code FROM stage_trial_balance_errors ORDER BY code, trans_id;
+  SELECT DISTINCT BUID(record_uuid) AS record_uuid, trans_id, account_id, account_label, code FROM stage_trial_balance_errors ORDER BY code, trans_id;
```
Comment (jniles) [inline server/models/procedures/trial_balance.sql]: Does `account_label` still exist?  It looks like it should be `error_description`.

File: server/models/procedures/trial_balance.sql [hunk 2]
```diff
@@ -96,91 +96,107 @@ CREATE PROCEDURE TrialBalanceErrors()
...
     FROM posting_journal AS pj JOIN stage_trial_balance_transaction AS temp
       ON pj.record_uuid = temp.record_uuid
+      JOIN account AS ac ON ac.id = pj.account_id
     WHERE pj.description IS NULL
     GROUP BY pj.record_uuid;
 
   -- check that all periods are unlocked
   INSERT INTO stage_trial_balance_errors
-    SELECT pj.record_uuid, MAX(pj.trans_id), 'POSTING_JOURNAL.ERRORS.LOCKED_PERIOD' AS code
+    SELECT pj.record_uuid, MAX(pj.trans_id), ac.id, CONCAT(ac.number,' - ', ac.label)  as error_description, 'POSTING_JOURNAL.ERRORS.LOCKED_PERIOD' AS code
```
Comment (jniles) [inline server/models/procedures/trial_balance.sql]: Let's just remove `ac.id` from all these.  It looks like we don't use it on the client anyway..

**Review by jniles — COMMENTED** [review]
Just a few clean up.  This looks good.

**Comment by jniles** [comment]
I'm gonna try to get this merged ahead of the upgrades.

**Comment by jniles** [comment]
Alright, got this working:

trialbalanceerrors
_Fig 1: New Errors Presentation_

Great work @jeremielodi !

bors r+

**Pull request merged**